src/Storages/Streaming/StreamingStoreSourceBase.h

```diff
@@ -50,7 +50,8 @@ class StreamingStoreSourceBase : public ISource
 
     Int64 last_sn = -1;
     Int64 last_epoch = -1;
-    /// FIXME, switch to llvm-15
-    std::atomic<CheckpointContext *> ckpt_ctx;
+    /// FIXME, switch to llvm-15 atomic shared_ptr
+    CheckpointContextPtr ckpt_ctx TSA_PT_GUARDED_BY(ckpt_mutex);
```

Comment:
1. In general (general guideline), `mutex` shall be defined before the data members it protects since when dtor, data members will be dtor first and then mutex and this is safer in different scenarios. 
2. There is race condition, this is exactly why we have a `FIXME`. Can you try if `std::atomic<CheckpointContextPtr>` works ? And for now, we tolerate this race because they are usually scheduled at least at least seconds if not mins away. So in practice, it may not quite matter. Having said this, can you try `std::atomic<CheckpointContextPtr>`